src/lib/cockpitversion.js

```diff
@@ -0,0 +1,2 @@
+const exec = require('child_process').execSync;
+module.exports=exec('git rev-parse --short HEAD').toString().trim();
```

Comment:
On first access the system will query git for the short hash of the current code and cache it for subsequent access.